### PR TITLE
[DOC] Finish adding badges to changelog

### LINKED ISSUE #3578: Apply changelog badges backwards
In #3542 we started adding badges to the changelog. This was applied backwards up to v0.9.0, but we wish to apply them fully to all the changelog files in `doc/changes` with the exception of `latest.rst` which will be handled during a release. This issue can be a reference for this. This can be done across several PRs by several contributors if it's too much to handle all at once. So anybody can take part by commenting to say which file(s) they will work on and open a pull request referencing this issue.

Badge mapping:

    :bdg-primary:`Doc`
    :bdg-secondary:`Maint`
    :bdg-success:`API`
    :bdg-info:`Plotting`
    :bdg-warning:`Test`
    :bdg-danger:`Deprecation`
    :bdg-dark:`Code`

Unassigned badge if a new category is warranted:

    :bdg-light:``

### Affected pages

Dev docs with some badges already added: https://nilearn.github.io/dev/changes/whats_new.html

See PR #3542 for a code reference on how badges are applied

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #3578

Changes proposed in this pull request:

- add badges to all badgeless entries in the changelog

### CODE REVIEW

**Comment by ymzayek** [comment]
A few missed ones:

```bash
doc/changes/0.2.2.rst:17:- Fetcher for Megatrawl Netmats dataset.
doc/changes/0.3.0.rst:36:- :term:`ANOVA` :term:`SVM` example on Haxby datasets ``plot_haxby_anova_svm`` in Decoding section now uses ``SelectPercentile`` to select :term:`voxels<voxel>` rather than ``SelectKBest``.
doc/changes/0.3.1.rst:16:- **Dropped support for scikit-learn older than 0.14.1** Minimum supported version is now ``0.15``.
doc/changes/0.4.0.rst:29:- Function :func:`~datasets.fetch_atlas_talairach` to fetch the `Talairach atlas <http://talairach.org>`_ (:gh:`1523` by `Jerome Dockes`_).
doc/changes/0.5.0.rst:239:- New :ref:`interactive plotting functions <interactive-plotting>`, eg for use in a notebook.
doc/changes/0.9.2.rst:67:- 64-bit integers in Nifti files: some tools such as FSL, SPM and AFNI cannot
```

**Comment by Remi-Gau** [comment]
> A few missed ones:
> 
> ```shell
> doc/changes/0.2.2.rst:17:- Fetcher for Megatrawl Netmats dataset.
> doc/changes/0.3.0.rst:36:- :term:`ANOVA` :term:`SVM` example on Haxby datasets ``plot_haxby_anova_svm`` in Decoding section now uses ``SelectPercentile`` to select :term:`voxels<voxel>` rather than ``SelectKBest``.
> doc/changes/0.3.1.rst:16:- **Dropped support for scikit-learn older than 0.14.1** Minimum supported version is now ``0.15``.
> doc/changes/0.4.0.rst:29:- Function :func:`~datasets.fetch_atlas_talairach` to fetch the `Talairach atlas <http://talairach.org>`_ (:gh:`1523` by `Jerome Dockes`_).
> doc/changes/0.5.0.rst:239:- New :ref:`interactive plotting functions <interactive-plotting>`, eg for use in a notebook.
> doc/changes/0.9.2.rst:67:- 64-bit integers in Nifti files: some tools such as FSL, SPM and AFNI cannot
> ```

good catch: will fix

File: doc/changes/0.2.2.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ It aims at preparing a renewal of the tutorials.
 New
 ---
 
-- Fetcher for Megatrawl Netmats dataset.
+- :bdg-dark:`Code` Fetcher for Megatrawl Netmats dataset.
```
Comment (ymzayek) [inline doc/changes/0.2.2.rst:17]: Maybe API here

File: doc/changes/0.4.0.rst [hunk 2]
```diff
@@ -26,7 +26,7 @@ Enhancements
 
 - :bdg-success:`API` Function :func:`~image.resample_img` now takes a linear resampling option (implemented by Joe Necus).
 
-- Function :func:`~datasets.fetch_atlas_talairach` to fetch the `Talairach atlas <http://talairach.org>`_ (:gh:`1523` by `Jerome Dockes`_).
+- :bdg-dark:`Code` Function :func:`~datasets.fetch_atlas_talairach` to fetch the `Talairach atlas <http://talairach.org>`_ (:gh:`1523` by `Jerome Dockes`_).
```
Comment (ymzayek) [inline doc/changes/0.4.0.rst:29]: API here too

**Review by ymzayek — APPROVED** [review]

File: doc/changes/0.2.2.rst (hunk 1 above)
Comment (Remi-Gau) [reply doc/changes/0.2.2.rst:17]: yup that makes more sense

File: doc/changes/0.4.0.rst (hunk 2 above)
Comment (Remi-Gau) [reply doc/changes/0.4.0.rst:29]: same

**Pull request merged**